.github/workflows/main.yml

```diff
@@ -99,7 +99,7 @@ jobs:
 
   linux-build:
     name: Build Linux binaries
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-24.04
```

Comment:
Why use 24.04 here, but 22.04 in the release workflow? imo it's best to build on the oldest kernel that's feasible. the resulting binaries will depend on the kernel's userspace abi, and care is taken to ensure that newer kernels support the abis from older kernels.

If the kernels used for the different workflows are intentionally different then it's probably worth commenting (in each workflow) why this is the case.